api/src/main/java/org/accula/api/code/CodeLoaderImpl.java

```diff
@@ -82,9 +89,59 @@
                 .flatMap(repo -> Mono.justOrEmpty(getObjectLoader(repo, commit.getSha(), filename)))
                 .switchIfEmpty(Mono.error(FILE_NOT_FOUND))
                 .map(loader -> cutFileContent(loader, fromLine, toLine))
+                .map(content -> new FileEntity(commit, filename, content))
                 .switchIfEmpty(Mono.error(CUT_ERROR));
     }
 
+    @Override
+    public Flux<Tuple2<FileEntity, FileEntity>> getDiff(final Commit base, final Commit head) {
+        final Mono<AbstractTreeIterator> baseTree = repositoryProvider
```

Comment:
So many code snippets like this in a whole file
```java
repositoryProvider
    .getRepository(base.getOwner(), base.getRepo())
```

Consider extracting this into `getRepo(Commit)`